tests/include/mock_custom_nrf_modem_at.h

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#ifndef MOCK_CUSTOM_NRF_MODEM_AT_H_
+#define MOCK_CUSTOM_NRF_MODEM_AT_H_
+
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @file mock_custom_nrf_modem_at.h
+ * @brief Public APIs for the Custom nrf_modem_at mock.
+ * @defgroup mock_custom_nrf_modem_at Custom mock for nrf_modem_at
+ * @{
+ */
+
+/* TODO: Add __wrap_nrf_modem_at_scanf_Ignore */
+
+/**
+ * @brief Initialize mocking variables for nrf_modem_at_scanf() calls.
+ *
+ * This is planned to be used in setUp() function.
+ */
+void mock_custom_nrf_modem_at_Init(void);
+/**
+ * @brief Verify that nrf_modem_at_scanf() calls where executed as expected.
+ *
+ * This is planned to be used in tearDown() function.
+ */
+void mock_custom_nrf_modem_at_Verify(void);
+
+/**
+ * @brief Mock for nrf_modem_at_scanf().
+ */
+int __wrap_nrf_modem_at_scanf(const char *cmd, const char *fmt, ...);
+
+/**
+ * @brief Set expected parameter and return values for non-variable arguments.
+ */
+void __wrap_nrf_modem_at_scanf_ExpectAndReturn(const char *cmd, const char *fmt, int return_value);
```

Comment:
I don't know what's preferred naming. An alternative is `__wrap_custom_nrf_modem_at_scanf_ExpectAndReturn()`

Reply:
I prefer `__wrap_custom` to keep it different from the other auto generated wrap functions.

Reply:
I've changed `__wrap_nrf_modem_at_scanf_*()` functions to `__wrap_custom_nrf_modem_at_scanf_*()`.
Notice that `__wrap_nrf_modem_at_scanf()` must remain as is in order for the mock to work and be found by the linker.

Reply:
What is the purpose of letting custom mocks look different from an auto-generated mock? Would it not be more desirable to have custom and autogenerated mocks look as similar as possible, so that the developers don't need to bother about that?

Reply:
Copying comment from @glarsennordic:
"On the flip side, this could also confuse developers, who might go looking for CMock documentation on the custom mocks and find none"

Reply:
There are pros and cons in either solution. While I'm open to both options, I do agree that having everything aligned with auto-generated stuff might be confusing as people start looking for CMock documentation and won't find `__wrap_nrf_modem_at_scanf_ReturnVarg_int16()` etc. functions.

Reply:
That's a fair point

Reply:
of course such a change is a bit out of scope for this specific PR, but if people believe it to be better to adjust CMock generated mocks, and thereby free up `__wrap` to hand written mocks, then such a PR can be opened in addition to this PR.

Reply:
Which teams chat was this mentioned in @trantanen? I think I am not invited to it

I'm fine with generated functions being called `__wrap_cmock_*` but it sounds like there could be a very long period of time when both these and normal cmock mocks are ***both*** called `__wrap_*`, and that I think will be confusing.

If we are planning to rename generated mocks to  `__wrap_cmock_*` why not also change from `__wrap_custom_*` to `__wrap_*` in ***that*** PR rather than this one?

Or, if `custom` really does need to be avoided from the start, let's at minimum pick a name that is distinct from how the generated mocks are named. What about `__wrap_variadic_*` or `__wrap_var_*` or `__fake_*` or `__wrap_nrfx_*` or `__wrap_nordic_*` or `__variadic_wrap_*` or `__var_wrap_*`?

Reply:
I prefer `__wrap_custom` or `stub_` . Something that differentiates these from the auto-generated ones is better.

Reply:
@balaji-nordic The idea is that ultimately generated will be `__wrap_cmock_*`, `__wrap_generated_*` etc. and mocks made by as are just `__wrap_*`. The question is what to do in the intermediate term.

Reply:
okay. I still think `__wrap_custom` is good prefix for the functions you added in the intermediate term.

Reply:
I'll be on vacation next week. It would be nice if you could find a common solution (@tejlmand, @balaji-nordic) so we could get this closed when I'm back.

Reply:
@tejlmand Are you alright with us just leaving it as `__wrap_custom_` and saving the renaming to `__wrap_` for the same PR where we rename all of the cmock wraps to `__wrap_cmock_`?

Reply:
Created: https://github.com/nrfconnect/sdk-nrf/pull/9283

Reply:
> Created: #9283

Let's get that PR merged first and then handle this PR.